### DISCUSSION TITLE
Index layout route

### DISCUSSION THREAD

**Discussion by xixixao** [remix-run/remix#6227] [Q&A] [answered]
I want a route to wrap the index page. I can't get it to work, despite following several examples online, and can't figure it out from docs. I'm getting:

```
Route ID Collision: "routes/__home"

The following routes all define the same Route ID, only the first one will be used

🟢 routes/__home/index.jsx
⭕️️ routes/__home.jsx
```

What part of https://remix.run/docs/en/1.15.0/guides/routing#pathless-layout-routes am I getting wrong?

I'm on 1.15

**Comment by xixixao** [comment]
And I guess related I don't understand why `npx create-remix@latest` creates a `_index.jsx` file, but the docs never mention `_index.jsx`, only `index.jsx`.

**Reply by sergiodxa** [reply, marked as answer]
Because it has [v2 route convention](https://remix.run/docs/en/1.15.0/file-conventions/route-files-v2) enabled by default, which is why your errors are not working too, in v2 convention you don’t use folders anymore.

I recommend you to either turn off v2 convention in your remix.config.js or read the whole guide about how it works since it changes everything about the file convention.

**Comment by eduardodallmannsysmap** [comment]
Hi everyone, I'm also facing this issue in my application when migrating to the latest versions of Remix. I use the double underscore layout file naming convention extensively. I'm using the createRoutesFromFolders function in remix config, and I'm still receiving many route collision warnings.

```
⚠️ Route ID Collision: "routes/__app"

The following routes all define the same Route ID, only the first one will be used

🟢 routes/__app/index.tsx
⭕️️ routes/__app.tsx
```

I believed that with 'createRoutesFromFolders' function I could use v1 file convention fully. Is there a way to avoid this warning without changing my naming convention?

Thanks for your help!

**Reply by sergiodxa** [reply]
I think if you set `future.v2_routeConvention = false` in your `remix.config.js` file it will hide the warning while retaining the v1 convention.

**Reply by eduardodallmannsysmap** [reply]
You are right. I set `v2_routeConvention` to true with the intention of omitting the warning `⚠️ REMIX FUTURE CHANGE: The route file convention is changing in v2`. My mistake.

**Reply by kiliman** [reply]
I think what you really want to do is to ignore all route files in remix.config. By default, Remix will scan your routes folder with whatever convention is operative. Then it checks if the `routes` option is defined and will call that function. So you end up with double-processing of routes.

This is what I would do:

```js
// remix.config.js
const {
  createRoutesFromFolders,
} = require("@remix-run/v1-route-convention");

/** @type {import('@remix-run/dev').AppConfig} */
module.exports = {
  future: {
    // makes the warning go away in v1.15
    v2_routeConvention: true,
  },
  // ignore all files in routes folder to prevent
  // default remix convention from picking up routes
  ignoredRouteFiles: ['**/*'],
  routes(defineRoutes) {
    // uses the v1 convention, works in v1.15+ and v2
    return createRoutesFromFolders(defineRoutes, 
      // optionally add the real ignored route files here as needed
      { ingnoredRouteFiles: ['...'] }
    );
  },
};
```

**Comment by cmnstmntmn** [comment]
still a problem 2 yers after.
